--- form.tpl	2017-04-17 12:02:25.000000000 +0000
+++ form.tpl.new	2017-04-17 12:11:15.020831376 +0000
@@ -12,7 +12,6 @@
   <link rel="stylesheet" type="text/css" href="/static/css/bootstrap.min.css">
   <link rel="stylesheet" type="text/css" href="/static/css/jquery-ui.min.css">
   <link rel="stylesheet" type="text/css" href="/static/css/bootstrap-select.min.css">
-  <link rel="stylesheet" type="text/css" href="/static/css/schoolhouse.css">
 
 </head>
 <body>
