--- qsi_ccd.cpp	2017-04-15 00:01:40.000000000 +0000
+++ qsi_ccd.cpp.new	2017-04-15 00:07:47.531271278 +0000
@@ -176,19 +176,19 @@
     IUFillSwitch(&FilterS[1], "FILTER_CCW", "-", ISS_OFF);
     IUFillSwitchVector(&FilterSP, FilterS, 2, getDeviceName(), "FILTER_WHEEL_MOTION", "Turn Wheel", FILTER_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
 
-    IUFillSwitch(&GainS[0], "High", "", ISS_OFF);
-    IUFillSwitch(&GainS[1], "Low", "", ISS_OFF);
-    IUFillSwitch(&GainS[2], "Auto", "", ISS_ON);
-    IUFillSwitchVector(&GainSP, GainS, 3, getDeviceName(), "Gain", "", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
+    IUFillSwitch(&GainS[0], "High", "High", ISS_OFF);
+    IUFillSwitch(&GainS[1], "Low", "Low", ISS_OFF);
+    IUFillSwitch(&GainS[2], "Auto", "Auto", ISS_ON);
+    IUFillSwitchVector(&GainSP, GainS, 3, getDeviceName(), "Gain", "Gain", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
 
-    IUFillSwitch(&FanS[0], "Off", "", ISS_OFF);
-    IUFillSwitch(&FanS[1], "Quiet", "", ISS_OFF);
-    IUFillSwitch(&FanS[2], "Full", "", ISS_ON);
-    IUFillSwitchVector(&FanSP, FanS, 3, getDeviceName(), "Fan", "", MAIN_CONTROL_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
+    IUFillSwitch(&FanS[0], "Off", "Off", ISS_OFF);
+    IUFillSwitch(&FanS[1], "Quiet", "Quiet", ISS_OFF);
+    IUFillSwitch(&FanS[2], "Full", "Full", ISS_ON);
+    IUFillSwitchVector(&FanSP, FanS, 3, getDeviceName(), "Fan", "Fan", MAIN_CONTROL_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
 
-    IUFillSwitch(&ABS[0], "Normal", "", ISS_ON);
-    IUFillSwitch(&ABS[1], "High", "", ISS_OFF);
-    IUFillSwitchVector(&ABSP, ABS, 2, getDeviceName(), "AntiBlooming", "", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
+    IUFillSwitch(&ABS[0], "Normal", "Normal", ISS_ON);
+    IUFillSwitch(&ABS[1], "High", "High", ISS_OFF);
+    IUFillSwitchVector(&ABSP, ABS, 2, getDeviceName(), "AntiBlooming", "AntiBlooming", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 60, IPS_IDLE);
 
     initFilterProperties(getDeviceName(), FILTER_TAB);
 
