--- default	2016-10-27 18:22:42.000000000 +0000
+++ default.new	2017-04-13 11:48:32.404501919 +0000
@@ -27,7 +27,7 @@
 	#
 	# include snippets/snakeoil.conf;
 
-	root /var/www/html;
+	root /home/pi/www;
 
 	# Add index.php to the list if you are using PHP
 	index index.html index.htm index.nginx-debian.html;
@@ -42,14 +42,14 @@
 
 	# pass the PHP scripts to FastCGI server listening on 127.0.0.1:9000
 	#
-	#location ~ \.php$ {
-	#	include snippets/fastcgi-php.conf;
-	#
+	location ~ \.php$ {
+		include snippets/fastcgi-php.conf;
+	
 	#	# With php5-cgi alone:
 	#	fastcgi_pass 127.0.0.1:9000;
-	#	# With php5-fpm:
-	#	fastcgi_pass unix:/var/run/php5-fpm.sock;
-	#}
+		# With php5-fpm:
+		fastcgi_pass unix:/var/run/php5-fpm.sock;
+	}
 
 	# deny access to .htaccess files, if Apache's document root
 	# concurs with nginx's one
